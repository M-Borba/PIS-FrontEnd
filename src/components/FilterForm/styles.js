import { makeStyles } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme) => ({
  paper: {
    margin: theme.spacing(8, 4),
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    gap: "1vw",
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));
