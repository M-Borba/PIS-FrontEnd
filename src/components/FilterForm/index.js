import React from "react";
import PropTypes from "prop-types";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import { useStyles } from "./styles";
import FilterAltIcon from "@mui/icons-material/FilterAlt";

FilterForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,

  project_type: PropTypes.string,
};

export default function FilterForm({ onSubmit, project_type }) {
  const classes = useStyles();

  return (
    <div className={classes.paper} data-testid="login">
      <Typography component="h1" variant="h5">
        Filtros
      </Typography>
      <form className={classes.form} noValidate onSubmit={onSubmit}>
        <TextField
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="project_type"
          type="text"
          label="tipo de proyecto"
          name="project_type"
          autoComplete="project_type"
          value={project_type}
          onChange={(e) => onInputChange(e)}
          autoFocus
        />
        <Button
          role="submit"
          type="submit"
          fullWidth
          variant="contained"
          color="primary"
          className={classes.submit}
        >
          Filtrar &ensp;
          <FilterAltIcon />
        </Button>

        <Box mt={5}></Box>
      </form>
    </div>
  );
}
