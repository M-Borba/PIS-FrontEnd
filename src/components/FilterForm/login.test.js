import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import FilterForm from "./index";
import React from "react";

describe("login component tests", () => {
  it("renders correctly", () => {
    const mockSubmit = jest.fn((e) => e.preventDefault());
    const { container } = render(
      <FilterForm onSubmit={(e) => mockSubmit(e)} />
    );
    // Matches the last snapshot taken
    expect(container).toMatchSnapshot();
    // calls both functions
    fireEvent(screen.getByRole("submit"), new MouseEvent("click"));

    expect(mockSubmit).toHaveBeenCalled();
  });
});
