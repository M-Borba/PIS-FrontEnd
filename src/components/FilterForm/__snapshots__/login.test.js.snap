// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login component tests renders correctly 1`] = `
<div>
  <div
    class="makeStyles-paper-1"
    data-testid="login"
  >
    <h1
      class="MuiTypography-root MuiTypography-h5"
    >
      Filtros
    </h1>
    <form
      class="makeStyles-form-2"
      novalidate=""
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
          data-shrink="true"
          for="project_type"
          id="project_type-label"
        >
          tipo de proyecto
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="project_type"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="project_type"
            name="project_type"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
            >
              <span>
                tipo de proyecto
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-3 MuiButton-containedPrimary MuiButton-fullWidth"
        role="submit"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Filtrar  
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FilterAltIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.25 5.61C6.27 8.2 10 13 10 13v6c0 .55.45 1 1 1h2c.55 0 1-.45 1-1v-6s3.72-4.8 5.74-7.39c.51-.66.04-1.61-.79-1.61H5.04c-.83 0-1.3.95-.79 1.61z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiBox-root MuiBox-root-8"
      />
    </form>
  </div>
</div>
`;
